packages/babel-types/src/validators/isReferenced.js

```diff
@@ -45,6 +45,15 @@ export default function isReferenced(
     case "PrivateName":
       return false;
 
+    // no: class { NODE() {} }
+    // yes: class { [NODE]() {} }
+    // no: class { foo(NODE) {} }
+    case "ClassMethod":
+    case "ClassPrivateMethod":
+    case "ObjectMethod":
+      if (parent.params.includes(node)) {
```

Comment:
May want to add that a comment that fall-through is intentional

Reply:
Done.